src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -2100,11 +2205,21 @@ ManageWorkerPool(WorkerPool *workerPool)
 		/* experimental: just to see the perf benefits of caching connections */
 		int connectionFlags = 0;
```

Comment:
Probably here we should have a method that takes execution and returns connection flags.